TLM/TLM/UI/Helpers/SerializableUIOptionBase.cs

```diff
@@ -9,34 +9,76 @@ namespace TrafficManager.UI.Helpers {
 
     public abstract class SerializableUIOptionBase<TVal, TUI> : ILegacySerializableOption
         where TUI : UIComponent {
-        //Data:
-        public SerializableUIOptionBase(string fieldName) {
+
+        /* Data: */
+        public SerializableUIOptionBase(string fieldName, Options.PersistTo scope) {
+
             ValueField = typeof(Options).GetField(fieldName, BindingFlags.Static | BindingFlags.Public);
             if (ValueField == null) {
-                throw new Exception($"{typeof(Options)}.{fieldName} does not exists");
+                throw new Exception($"{typeof(Options)}.{fieldName} does not exist");
             }
+
+            Scope = scope;
         }
+
+        public TranslatorDelegate Translator;
         private FieldInfo ValueField;
-        private Options OptionInstance => Singleton<Options>.instance;
+        public Options.PersistTo Scope { get; private set; }
+
+        /// <summary>Gets or sets the value of the field this option represents.</summary>
```

Comment:
Now that I think about this, this could be major performance issue.

Reply:
its my fault though. this could be fix using harmony struct ref

Reply:
I've never used harmony so would need some pointers as to how to achieve that. Even then I assume perf critical code would still get even faster performance from just going direct to the option in `Options.cs`?